Dockerfile.boringssl

```diff
@@ -5,11 +5,11 @@
 # CLEAR_LINUX_VERSION="--version=29970"
 #
 # This is used on release branches before tagging a stable version.
-ARG CLEAR_LINUX_BASE=clearlinux:base
+ARG CLEAR_LINUX_BASE=clearlinux@sha256:b8e5d3b2576eb6d868f8d52e401f678c873264d349e469637f98ee2adf7b33d4
 
 FROM ${CLEAR_LINUX_BASE} as builder
 
-ARG CLEAR_LINUX_VERSION=
+ARG CLEAR_LINUX_VERSION="--version=29970"
```

Comment:
Hmm, what's special about this version?

Reply:
Without the env variable, the build will fail.
In my opinion, the version of all components should be fixed.
This can avoid some compilation problems caused by new version in the future.

Reply:
It might be better to submit these modifications in another patch.

Reply:
I would rather find out why the build fails with the latest version -- we should be able to target the latest Clear Linux.

Reply:
@derekguo001 the build has failures with GCC11. Can you update this PR so that the latest Clear Linux version is used but GCC10 is used (from `c-extras-gcc10` bundle)

Reply:
@mythi Hi, Mikko, I tried c-extras-gcc10 bundle with clearlinux:base, but GCC10 could not build Envoy correctly.

The modification in Dockerfile.boringssl:
`RUN swupd bundle-add os-core-dev c-extras-gcc10 python3-basic`

Compile Error:
`
Use --sandbox_debug to see verbose messages from the sandbox
external/boringssl/src/crypto/curve25519/curve25519.c:503:57: error: argument 2 of type 'const uint8_t[32]' {aka 'const unsigned ch
ar[32]'} with mismatched bound [-Werror=array-parameter=]
  503 | int x25519_ge_frombytes_vartime(ge_p3 *h, const uint8_t s[32]) {
      |                                           ~~~~~~~~~~~~~~^~~~~
In file included from external/boringssl/src/crypto/curve25519/curve25519.c:33:
external/boringssl/src/crypto/curve25519/internal.h:109:58: note: previously declared as 'const uint8_t *' {aka 'const unsigned cha
r *'}
  109 | int x25519_ge_frombytes_vartime(ge_p3 *h, const uint8_t *s);
      |                                           ~~~~~~~~~~~~~~~^
external/boringssl/src/crypto/curve25519/curve25519.c:823:57: error: argument 2 of type 'const uint8_t *' {aka 'const unsigned char
 *'} declared as a pointer [-Werror=array-parameter=]
  823 | void x25519_ge_scalarmult_base(ge_p3 *h, const uint8_t *a) {
      |                                          ~~~~~~~~~~~~~~~^
In file included from external/boringssl/src/crypto/curve25519/curve25519.c:33:
external/boringssl/src/crypto/curve25519/internal.h:117:56: note: previously declared as an array 'const uint8_t[32]' {aka 'const u
nsigned char[32]'}
  117 | void x25519_ge_scalarmult_base(ge_p3 *h, const uint8_t a[32]);
      |                                          ~~~~~~~~~~~~~~^~~~~
cc1: all warnings being treated as errors
Target //source/exe:envoy-static failed to build
`

Reply:
@derekguo001 

> The modification in Dockerfile.boringssl:
> `RUN swupd bundle-add os-core-dev c-extras-gcc10 python3-basic`

I don't think this alone is not enough. You'd need to set CC to be gcc10 for Envoy build.

Reply:
Sorry. I replaced c-basic with c-extras-gcc10 and did not notice that os-core-dev also contained another gcc11.
It's ok now.